web/frontend/src/app.vue

```diff
@@ -1,906 +1,45 @@
 
-import { onMounted } from 'vue';
-import { grpc } from '@improbable-eng/grpc-web';
-import { Duration } from 'google-protobuf/google/protobuf/duration_pb';
-import type { Credentials } from '@viamrobotics/rpc';
-import { ConnectionClosedError } from '@viamrobotics/rpc';
-import { toast } from './lib/toast';
-import { displayError } from './lib/error';
-import { addResizeListeners } from './lib/resize';
-import {
-  Camera,
-  CameraClient,
-  Client,
-  ResponseStream,
-  ServiceError,
-  commonApi,
-  robotApi,
-  sensorsApi,
-} from '@viamrobotics/sdk';
-
-import {
-  resourceNameToSubtypeString,
-  resourceNameToString,
-  filterResources,
-  filterNonRemoteResources,
-  filterRdkComponentsWithStatus,
-  filterComponentsWithNames,
-} from './lib/resource';
-
-import Arm from './components/arm.vue';
-import AudioInput from './components/audio-input.vue';
-import Base from './components/base.vue';
-import Board from './components/board.vue';
-import CameraView from './components/camera.vue';
-import OperationsSessions from './components/operations-sessions.vue';
-import DoCommand from './components/do-command.vue';
-import Gantry from './components/gantry.vue';
-import Gripper from './components/gripper.vue';
-import Gamepad from './components/gamepad.vue';
-import InputController from './components/input-controller.vue';
-import Motor from './components/motor-detail.vue';
-import MovementSensor from './components/movement-sensor.vue';
-import Navigation from './components/navigation.vue';
-import ServoComponent from './components/servo.vue';
-import Sensors from './components/sensors.vue';
-import Slam from './components/slam.vue';
-
-import {
-  fixArmStatus,
-  fixBoardStatus,
-  fixGantryStatus,
-  fixInputStatus,
-  fixMotorStatus,
-  fixServoStatus,
-} from './lib/fixers';
-
-const relevantSubtypesForStatus = [
-  'arm',
-  'gantry',
-  'board',
-  'servo',
-  'motor',
-  'input_controller',
-];
-
-const password = $ref<string>('');
-const supportedAuthTypes = $computed(() => window.supportedAuthTypes);
-const rawStatus = $ref<Record<string, robotApi.Status>>({});
-const status = $ref<Record<string, robotApi.Status>>({});
-const errors = $ref<Record<string, boolean>>({});
-
-let statusStream: ResponseStream<robotApi.StreamStatusResponse> | null = null;
-let lastStatusTS: number | null = null;
-let disableAuthElements = $ref(false);
-let cameraFrameIntervalId = $ref(-1);
-let currentOps = $ref<{ op: robotApi.Operation.AsObject, elapsed: number }[]>([]);
-let currentSessions = $ref<robotApi.Session.AsObject[]>([]);
-let sensorNames = $ref<commonApi.ResourceName.AsObject[]>([]);
-let resources = $ref<commonApi.ResourceName.AsObject[]>([]);
-let resourcesOnce = false;
-let errorMessage = $ref('');
-let connectedOnce = $ref(false);
-let connectedFirstTimeResolve: (value: void) => void;
-const connectedFirstTime = new Promise<void>((resolve) => {
-  connectedFirstTimeResolve = resolve;
-});
+const { host } = window;
```

Comment:
nit: i'd recommend this destructuring pattern:
```suggestion
const {
  bakedAuth,
  supportedAuthTypes,
  webrtcAdditionalICEServers,
  webrtcEnabled,
  webrtcSignalingAddress,
} = window;

Reply:
Yeah whoops good call :)

Reply:
Going to make one more change that I just realized as I'm cleaning up things. On the consumer side of the component, its better to allow for an optional `baked-auth` prop. Less type errors and annoyances that way. Will commit these changes now.